src/kernels/execution/kernelExecution.ts

```diff
@@ -32,134 +34,58 @@ import { noop } from '../../platform/common/utils/misc';
  * Separate class that deals just with kernel execution.
  * Else the `Kernel` class gets very big.
  */
-export class KernelExecution implements IDisposable {
-    private readonly documentExecutions = new WeakMap<NotebookDocument, CellExecutionQueue>();
-    private readonly executionFactory: CellExecutionFactory;
-    private readonly disposables: IDisposable[] = [];
+export class BaseKernelExecution<TKernel extends IBaseKernel = IBaseKernel> implements IDisposable {
```

Comment:
Could the base class live in a different file? Not sure if that's preferable or not

Reply:
Can, sometimes i find it useful to be in one place, else theres a lot of going back and forth..
However Will change

Reply:
Damn it, forgot this, will do in debt week